pennylane/qchem/convert.py

```diff
@@ -354,3 +355,264 @@ def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e01
+
+    for i in range(electrons):
+        singles_p += [i]
+        doubles_pq += [[k, i] for k in range(i)]
+    for j in range(electrons, orbitals):
+        singles_q += [j]
+        doubles_rs += [[k, j] for k in range(electrons, j)]
+
+    singles = [[p] + [q] for p in singles_p for q in singles_q]
+    doubles = [pq + rs for pq in doubles_pq for rs in doubles_rs]
+
+    return singles, doubles
+
+
+def _excited_configurations(electrons, orbitals, excitation):
+    r"""Generate excited configurations from a Hartree-Fock reference state.
+
+    This function generates excited configurations in the form of integers representing a binary
+    string, e.g., :math:`|1 1 0 1 0 0 \rangle is represented by :math:`int('110100', 2) = 52`.
+
+    The excited configurations are generated from a Hartree-Fock (HF) reference state. The HF state
+    is assumed to have the form :math:`|1 1 ...1 0 ... 0 0 \rangle` where the number of :math:`1`
+    and :math:`0` elements are the number of occupied and unoccupied spin orbitals, respectively.
+    The string representation of the state is obtained by converting the occupation-number vector to
+    a string, e.g., ``111000`` to represent :math:`|1 1 1 0 0 0 \rangle.
+
+    Each excited configuration has a sign, :math:`+1` or :math:`-1`, that is obtained by reordering
+    the creation operators.
+
+    Args:
+        electrons (int): number of electrons
+        orbitals (int): number of spin orbitals
+        excitation (int): number of excited electrons
+
+    Returns:
+        tuple(array, array): arrays of excited configurations and signs obtained by reordering the
+         creation operators
+
+    **Example**
+
+    >>> electrons = 2
+    >>> orbitals = 5
+    >>> excitation = 2
+    >>> _excitated_states(electrons, orbitals, excitation)
+    (array([28, 26, 25]), array([ 1, -1,  1]))
+    """
+    hf_state = qml.qchem.hf_state(electrons, orbitals)
+    singles, doubles = _excitations(electrons, orbitals)
+    states, signs = [], []
+
+    if excitation == 1:
+        for s in singles:
+            state = hf_state.copy()
+            state[s] = state[s[::-1]]
+            states += [state]
+            signs.append((-1) ** len(range(s[0], electrons - 1)))
+
+    if excitation == 2:
+        for d in doubles:
+            state = hf_state.copy()
+            state[d] = state[[d[2], d[3], d[0], d[1]]]
+            states += [state]
+            order_pq = len(range(d[0], electrons - 1))
+            order_rs = len(range(d[1], electrons - 1))
+            signs.append((-1) ** (order_pq + order_rs + 1))
+
+    states_str = ["".join([str(i) for i in state]) for state in states]
+    states_int = [int(state[::-1], 2) for state in states_str]
+
+    return states_int, signs
+
+
+def _ucisd_state(cisd_solver, tol=1e-15):
+    r"""Construct a wavefunction from PySCF's `UCISD` solver object.
+
+    The generated wavefunction is a dictionary where the keys represent a configuration, which
```

Comment:
Im not sure if we do this for doc-strings, but should we add a link to the PySCF documentation for this object?

Reply:
We can address this in a future PR dedicated to documentation.

Reply:
👍🏼